src/qt/nametablemodel.cpp

```diff
@@ -98,7 +103,156 @@ class NameTablePriv
                 std::string data = maybeData.get_str();
                 int height = find_value ( v, "height").get_int();
                 int expiresIn = find_value ( v, "expires_in").get_int();
-                vNamesO[name] = NameTableEntry(name, data, height, expiresIn, "confirmed");
+
+                bool isMine = find_value ( v, "ismine").get_bool();
+                bool isExpired = find_value ( v, "expired").get_bool();
+                // TODO: Check "op" field
+
+                std::string status = NameTableEntry::NAME_STATUS_CONFIRMED;
+                if (isExpired)
+                {
+                    status = NameTableEntry::NAME_STATUS_EXPIRED;
+                }
+                if (!isMine)
+                {
+                    status = NameTableEntry::NAME_STATUS_TRANSFERRED_OUT;
+                }
+
+                vNamesO[name] = NameTableEntry(name, data, height, expiresIn, status);
+            }
+        }
+
+        // unconfirmed names (name_pending)
+        // TODO: Set name and value encoding to hex, so that nonstandard
+        // encodings don't cause errors.
+
+        UniValue pendingNames;
+        try {
+            pendingNames = parent.walletModel->node().executeRpc("name_pending", NullUniValue, walletURI);
+        } catch (const UniValue& e) {
+            // although we shouldn't typically encounter error here, we
+            // should continue and try to add confirmed names and
+            // pending names. show error to user in case something
+            // actually went wrong so they can potentially recover
+            UniValue message = find_value( e, "message");
+            LogPrintf ("name_pending lookup error: %s\n", message.get_str());
+        }
+
+        // will be an object if name_pending command isn't available/other error
+        if(pendingNames.isArray())
+        {
+            for (const auto& v : pendingNames.getValues())
+            {
+                UniValue maybeName = find_value ( v, "name");
+                UniValue maybeData = find_value ( v, "value");
+                if (!maybeName.isStr() || !maybeData.isStr())
+                {
+                    continue;
+                }
+
+                std::string name = maybeName.get_str();
+                std::string data = maybeData.get_str();
+
+                bool isMine = find_value ( v, "ismine").get_bool();
+                std::string op = find_value ( v, "op").get_str();
+
+                auto confirmedEntryExists = vNamesO.count(name);
+
+                int height = 0;
+                int expiresIn = 0;
+                std::string status;
+
+                if (!confirmedEntryExists)
```

Comment:
Optional nit:  I think it helps code readability in general if negative conditions are avoided as much as possible.  In this case, you have both an "if" and an "else" clause.  So I would suggest to swap the two, so that the if is for the positive `confirmedEntryExists`.